### ISSUE TITLE
MACRO redefintion on windows build

### ISSUE THREAD

**Issue by reid-p** [analogdevicesinc/libiio#425]
Just started getting a warning from MSVC build of libiio that the ERROR macro is redefined.

This started with the "coverity: fix contex.c Untrusted value as argument" commit.
Specifically debug.h defines an ERROR macro, and now that context.c include "network.h" which 
include the windows winsock headers which for me are include wingdi.h which also define an ERROR macro things don't build.

```
wingdi.h:
#define ERROR               0

debug.h
#  define ERROR(...) \
    fprintf(stderr, "ERROR: " __VA_ARGS__)
```

Thoughts?

**Comment by pcercuei** [comment]
I use the same debug.h in other projects, and at some point I had to rename the macros as I was facing the same problem. Names like 'ERROR' or 'DEBUG' are too common for macros. I renamed them to pr_debug, pr_info, pr_warn and pr_err.

**Comment by rgetz** [comment]
Ok - I will do the same here...

**Comment by rgetz** [comment]
Should be fixed when #427 is committed to master. If you have any more issues - please let us know, and re-open.
